src/components/RouteElevation.js

```diff
@@ -0,0 +1,139 @@
+import PropTypes from 'prop-types';
+import React, {Component} from 'react';
+import cheapRuler from 'cheap-ruler';
+import isEqual from 'lodash/isEqual';
+import {AreaChart, Area, YAxis} from 'recharts';
+
+class RouteElevation extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      status: 'pending'
+    };
+  }
+
+  render() {
+    switch (this.state.status) {
+    case 'pending':
+      return <div className='loading--s'/>;
+
+    case 'ok': {
+      let upsAndDowns = this.upsAndDowns();
+      return (
+        <div>
+        <div className='txt-s color-darken50'>
+          <svg className='icon inline-block align-middle'><use xlinkHref='#icon-arrow-up'></use></svg>{upsAndDowns[0]}m -
+          <svg className='icon inline-block align-middle'><use xlinkHref='#icon-arrow-down'></use></svg>{upsAndDowns[1]}m
+        </div>
+        <AreaChart width={420} margin={{top: 12, right: 42, left: 42, bottom: 12}} height={100} data={this.state.elevations.map(e => ({e: e}))}>
+          <YAxis orientation="right"/>
+          <Area type="monotone" dataKey="e" stroke="#2abaf7" fill='#2abaf7' fillOpacity={0.5} strokeWidth={2} dot={null} />
+        </AreaChart>
+        </div>
+      );
+    }
+
+    default:
+      return null;
+    }
+  }
+
+  componentWillReceiveProps(nextProps) {
+    // Usually asynchronous calls would happen in the API caller,
+    // but the results here are independent from the apps' state.
+
+    // First compare props to see if updating the state is necessary
+    if (isEqual(nextProps.route, this.props.route)) return null;
```

Comment:
I'm leaving this as is for now. Feel free to open a PR, even small - as I'm not too familiar with the component lifecycle?

Thanks for following along so reactively :)